batch/batch/worker/worker.py

```diff
@@ -1664,6 +1665,7 @@ async def run(self):
                     async with self.worker.jar_download_locks[self.revision]:
                         local_jar_location = f'/hail-jars/{self.revision}.jar'
                         if not os.path.isfile(local_jar_location):
+                            self.verify_is_acceptable_query_jar_url(self.jar_url)
                             async with await self.worker.fs.open(self.jar_url) as jar_data:
                                 await self.worker.fs.makedirs('/hail-jars/', exist_ok=True)
                                 async with await self.worker.fs.create(local_jar_location) as local_file:
```

Comment:
We need to shield the **entire** download operation here. See the change we had to make when downloading root fs.

Reply:
I still think you need to shield it if multiple jobs might try and download the same jar version.

Reply:
This is where the /exec not found bug was.